ddtrace/ext/tags.go

```diff
@@ -11,6 +11,9 @@ const (
 	// TargetHost sets the target host address.
 	TargetHost = "out.host"
 
+	// NetworkTargetHost is the updated hostname tag for outbound connections
+	NetworkTargetHost = "network.destination.name"
```

Comment:
```suggestion
        // NetworkDestinationName is the remote hostname or similar where the outbound connection is being made to.
	NetworkDestinationName = "network.destination.name"
```

2 suggestions:
- Keep the name of the constant same (or at least similar) to the tag name to avoid confusion.
- Use same description as the UNC one.

Reply:
Also I think we could probably mark the other tags (`out.host` and `peer.hostname`) with a `// Deprecated: use NetworkDestinationName instead.` comment, as it's currently very confusing to have 3 different tags for the same purpose.